scriptworker/gpg.py

```diff
@@ -1137,130 +1136,60 @@ def rebuild_gpg_home_signed(context, real_gpg_home, my_pub_key_path,
-        commit 6efb4ebe8900ad1920f6eaaf64b615fe6e6e839a
-        gpg: Signature made Mon Sep 19 21:50:53 2016 PDT
-        gpg:                using RSA key FC829B7FFAA9AC38
-        gpg: checking the trustdb
-        gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
-        gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
-        gpg: next trustdb check due at 2018-09-17
-        gpg: Good signature from "Aki Sasaki (2016.09.16) <aki@escapewindow.com>" [ultimate]
-        gpg:                 aka "Aki Sasaki (2016.09.16) <aki@mozilla.com>" [ultimate]
-        gpg:                 aka "Aki Sasaki (2016.09.16) <asasaki@mozilla.com>" [ultimate]
-        gpg:                 aka "[jpeg image of size 5283]" [ultimate]
-        Author: Aki Sasaki <aki@escapewindow.com>
-        Date:   Mon Sep 19 21:50:35 2016 -0700
-
-            add another check + small fixes + comments
-
-    or::
-
-        commit 02dc29251021519ebac4508545477a7b23efea49
-        gpg: Signature made Tue Sep 20 04:22:57 2016 UTC
-        gpg:                using RSA key 0xFC829B7FFAA9AC38
-        gpg: Good signature from "Aki Sasaki (2016.09.16) <aki@escapewindow.com>"
-        gpg:                 aka "Aki Sasaki (2016.09.16) <aki@mozilla.com>"
-        gpg:                 aka "Aki Sasaki (2016.09.16) <asasaki@mozilla.com>"
-        gpg:                 aka "[jpeg image of size 5283]"
-        Author: Aki Sasaki <aki@escapewindow.com>
-        Date:   Mon Sep 19 21:22:40 2016 -0700
-
-            add travis tests for commit signatures.
-
+async def get_git_revision(path, ref="HEAD",
+                           exec_function=asyncio.create_subprocess_exec):
+    """Get the git revision of path.
 
     Args:
-        output (str): the output from ``git log --no-merges -n 1 --show-signature``
+        path (str): the path to run ``git log -n1 --format=format:%H REF`` in.
+        ref (str, optional): the ref to find the revision for.  Defaults to "HEAD"
+
+    Returns:
+        str: the revision found.
 
     Raises:
-        ScriptWorkerGPGException: on error.
+        ScriptWorkerRetryException: on failure.
     """
-    BAD = {
-        "gpg: Can't check signature:": "",
-        "gpg: WARNING: This key is not certified with a trusted signature!": "",
-    }
-    GOOD = re.compile(r"""^gpg: Good signature from ".*"( \[(ultimate|trusted)\])?$""")
-    messages = []
-    lines = output.splitlines()
-    status = False
-    for line in lines:
-        m = GOOD.match(line)
-        if m is not None:
-            status = True
-            continue
-        for k, v in BAD.items():
-            if line.startswith(k):
-                messages.append(v)
-                continue
-    if not status:
-        messages.append("No trusted signature!")
-    if messages:
-        raise ScriptWorkerGPGException("\n".join(messages + ["output:", output]))
+    proc = await exec_function(
+        'git', "log", "-n1", "--format=format:%H", ref, cwd=path,
```

Comment:
I wonder `git show-ref` doesn't pick the most recent HEAD across every branch. The documentation isn't clear about that. Thanks for pointing out the issue!